src/AppInstallerRepositoryCore/RepositorySearch.cpp

```diff
@@ -105,6 +105,22 @@ namespace AppInstaller::Repository
         return Version.empty() && Channel.empty();
     }
 
+    PackageVersionMultiProperty PackageMultiPropertyToPackageVersionMultiProperty(PackageMultiProperty property)
+    {
+        switch (property)
+        {
+        case PackageMultiProperty::PackageFamilyName: return PackageVersionMultiProperty::PackageFamilyName;
+        case PackageMultiProperty::ProductCode: return PackageVersionMultiProperty::ProductCode;
+        case PackageMultiProperty::UpgradeCode: return PackageVersionMultiProperty::UpgradeCode;
+        case PackageMultiProperty::NormalizedName: return PackageVersionMultiProperty::Name;
+        case PackageMultiProperty::NormalizedPublisher: return PackageVersionMultiProperty::Publisher;
+        case PackageMultiProperty::Tag: return PackageVersionMultiProperty::Tag;
+        case PackageMultiProperty::Command: return PackageVersionMultiProperty::Command;
+        default:
+            THROW_HR(E_UNEXPECTED);

```

Comment:
I personally get quite frustrated whenever I see a program give E_UNEXPECTED. Perhaps E_INVALIDARG?

Reply:
My expectation is that this error can only be produced by an error on the part of the developer, and that said error should be evident to the developer before it ever makes it to PR.  Invalid argument makes no more sense than unexpected if the exception makes it to the end user.  In fact, it can be more confusing because they might think that it was their fault.

I've added a message at least.